backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,178 @@
+
+  private static final String CREDENTIALS_FILE_PATH = "backend/credentials.json";
+
+  /** The email to be notified */
+  private final String toEmail;
+
+  /** The notification recipient */
+  private final String recipientName;
+
+  /**
+   * @param recipientName Name of the recipient of the user
+   * @param toEmail The email address that this email is going to be sent to.
+   */
+  public EmailNotifier(String recipientName, String toEmail) {
+    //    if (gmail == null) {
+    //      throw new InvalidParameterException("Gmail Cannot be Null");
+    //    }
+    this.recipientName = recipientName;
+    this.toEmail = toEmail;
+  }
+
+  /**
+   * Creates an authorized Credential object.
+   *
+   * @param HTTP_TRANSPORT The network HTTP Transport.
+   * @return An authorized Credential object.
+   * @throws IOException If the credentials.json file cannot be found.
+   */
+  public static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+      throws IOException {
+    // Load client secrets.
+    InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH);
+    GoogleClientSecrets clientSecrets =
+        GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+    // Build flow and trigger user authorization request.
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+            .setDataStoreFactory(new FileDataStoreFactory(new File(TOKENS_DIRECTORY_PATH)))
+            .setAccessType("offline")
+            .build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8000).build();
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  /**
+   * Create a message from an email.
+   *
+   * @param emailContent Email to be sent to raw of message
+   * @return a message containing a base64url encoded email
+   * @throws IOException if an error occurs writing to the stream
+   * @throws MessagingException for other failures
+   */
+  public static Message createMessageWithEmail(MimeMessage emailContent)
+      throws MessagingException, IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    emailContent.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  /**
+   * Create a MimeMessage using the parameters provided.
+   *
+   * @param toEmail email address of the receiver
+   * @param subject subject of the email
+   * @param bodyText body text of the email
+   * @return the MimeMessage to be used to send email
+   * @throws MessagingException if there was a problem accessing the Store
+   */
+  public MimeMessage createEmail(String toEmail, String subject, String bodyText)
```

Comment:
Since it's not a static function anymore, there's no need to pass `toEmail`, as it's a property of the object.